fpdf/fpdf.py

```diff
@@ -4015,15 +4066,56 @@ def _putfonts(self):
                 self._out("endobj")
             elif my_type == "TTF":
                 self.fonts[font_name]["n"] = self.n + 1
-                ttf = TTFontFile()
                 fontname = f"MPDFAA+{font['name']}"
                 subset = font["subset"].dict()
+
+                # why we delete 0-element?
                 del subset[0]
-                ttfontstream = ttf.makeSubset(font["ttffile"], subset)
+
+                # ---- FONTTOOLS SUBSETTER ----
+                # recalcTimestamp=False means that it doesn't modify the "modified" timestamp in head table
+                # if we leave recalcTimestamp=True the tests will break every time
+                ft = ttLib.TTFont(file=font["ttffile"], recalcTimestamp=False)
+
+                # 1. get all glyphs in PDF
+                cmap = ft["cmap"].getBestCmap()
+                glyph_names = [cmap[code] for code in subset if code in cmap]
+
+                # 2. make a subset
+                # notdef_outline=True means that keeps the white box for the .notdef glyph
+                # recommended_glyphs=True means that adds the .notdef, .null, CR, and space glyphs
+                options = ftsubset.Options(notdef_outline=True, recommended_glyphs=True)
```

Comment:
Fonts contain many tables not needed in PDF (e.g. GSUB, GPOS), you might want to check the tables in the font and ask the subsetter to drop anything not needed in PDF.

Reply:
Yes I agree that it would be better to drop everything not used in the PDF, do you know some source references where I can find the tables used in PDF?

Reply:
No, unfortunately. You can at least drop GDEF/GSUB/GPOS tables, these are usually some of the biggest and are also the slowest to subset.

Reply:
Okay thank you, for now I decided to keep only the tables that were kept in the old code